EIPS/eip-7425.md

```diff
@@ -0,0 +1,177 @@
+	*/
+	function proposalDeposit(uint256 assets, address receiver, uint256 proposal) external virtual returns(uint256);
+	/**
+	* @dev Burn shares, receive 1 to 1 value of shares
+	* - MUST have closed proposalNumber
+	* - MUST have userDeposit greater than or equal to userWithdrawal
+	* @param assets Amount being deposited
+	* @param receiver Address of receiver
+	* @param owner Address of token owner
+	* @param proposal Number associated proposal
+	*/
+	function proposalWithdraw(uint256 assets, address receiver, address owner, uint256 proposal)external virtual returns(uint256);
+	/**
+	* @dev Issue new proposal
+	* - MUST create new proposal number
+	* - MUST account for amount withdrawn
+	* - MUST emit proposals event
+	* @param token Address of ERC-20 token
+	* @param amount Token amount being withdrawn
+	* @param receiver Address of token recipient
+	*/
+	function proposalOpen(address token, uint256 amount, address receiver) external virtual returns (uint256);
+	/**
+	* @dev Close an opened proposal
+	* - MUST account for amount received
+	* - MUST be a proposal that is less than or equal to current proposal
+	* - MUST emit proposals event
+	* @param token Address of ERC-20 token
+	* @param proposal Number of desired proposal to close
+	* @param amount Number assets being received
+	*/
+	function proposalClose(address token, uint256 proposal, uint256 amount) external virtual returns (bool);
+	/**
+	* @dev Optional accounting for tokens deposited by owner
+	* - MUST be reserve owner
+	* - MUST emit depositR event
+	* NOTE: No shares are issued, funds can not be redeemed. Only withdrawn from proposalOpen
+	* @param token Address of ERC-20 token
+	* @param sender Address of where tokens from
+	* @param amount Number of assets being deposited
+	*/
+	function depositReserve(address token, address sender, uint256 amount) external virtual;
+}
+    
+```
+
+## Rationale
+
+The reserve is designed to be a basic implementation of the reserve interface. Other non specified conditions SHOULD be addressed on case by case basis. Reserves use [ERC-20](../EIPS/eip-20.md) as the standard for shares, and [ERC-4626](../EIPS/eip-4626.md) for the creation of shares. Reservse tokens can be considered as the underlying token in [ERC-4626](../EIPS/eip-4626.md) standard or the shares that get created when the underlying token is deposited in the vault. 
+[ERC-4626](../EIPS/eip-4626.md) is implemented to the reserve to account for user participation. There MUST be a representation of participation in a proposal for the user within a reserve. A user MUST NOT use the same tokens in multiple proposals. Once the underlying token is deposited into the vault for a open proposal those tokens are not accessible until a proposal is closed. It is not explicitly enforced that deposited tokens that create shares MUST NOT be withdrawn by the owner of the reserve. On a case by case basis there SHOULD be measures to ensure those tokens are accounted for if needed.  
```

Comment:
The RFC 2119 keywords create "requirements" on implementations, and you shouldn't introduce new requirements in the rationale section. They should go in the specification section.

For example, this should go in the specification section:

>  A user MUST NOT use the same tokens in multiple proposals.

And something like this would go in the rationale section:

> A user cannot use the same tokens in multiple proposals because [it will release a dragon.]